### PR TITLE
test: Look at averages when checking metrics

### PR DESCRIPTION
**Pull request by mvollmer**
Otherwise any short outlier will make us miss the plateau.  Such
outliers happen frequently with the CPU plot.

### CODE REVIEW

File: test/verify/check-metrics [hunk 1]
```diff
@@ -54,25 +54,25 @@ class TestMetrics(MachineCase):
...
-                  start = data[i][0];
-                  start_index = i;
-                  sum = 0;
-                }
-                sum += data[i][1];
-                if (data[i][0] - start >= duration * 1000) {
-                  console.log("avg", label, sum/(i - start_index + 1));
-                  return true;
+              sum = 0;
+              for (j = i; j < data.length; j++) {
```
Comment (dperpeet) [inline test/verify/check-metrics]: - This is probably the most inefficient way of calculating a moving average. The sum shouldn't be calculated completely for each sub-range.
- The inner loop could check if all the remaining samples cover a window large enough before even starting to calculate (time delta between current begin `i` and last sample `data[data.length-1]`)
- Indentation should be 4 spaces

File: test/verify/check-metrics [hunk 2]
```diff
@@ -88,10 +88,10 @@ class TestMetrics(MachineCase):
         # CPU.  We load one cpu for 20 seconds.  The resulting CPU
         # load can be all over the place since we don't control the
         # environment where the test virtual machine runs well enough.
-        # So we are happy with anything above 50% for 15 seconds.
+        # So we are happy with anything averaging more than 80% for 15 seconds.
```
Comment (dperpeet) [inline test/verify/check-metrics]: Should we add a comment that 100% equals one cpu fully loaded?

File: test/verify/check-metrics (hunk 1 above)
Comment (mvollmer) [inline test/verify/check-metrics]: Yes, it's O(n^2), but does it matter?  (I know these things come back to hunt one later...)

Actually, the j loop needs to end whenever the window is large enough.

Comment (mvollmer) [inline test/verify/check-metrics]: Fixed indentation.

Comment (mvollmer) [inline test/verify/check-metrics]: Changed to a moving average.

File: test/verify/check-metrics (hunk 2 above)
Comment (mvollmer) [inline test/verify/check-metrics]: Done.

**Comment by petervo** [comment]
Metrics test failed on fedora-atomic:

https://fedorapeople.org/groups/cockpit/logs/pull-3885-836936bb-verify-fedora-atomic/log.html#22

**Comment by mvollmer** [comment]
> Metrics test failed on fedora-atomic:

Yep.  Alright, looks like we need to measure the actual load in some other way and then check whether the plot reflects it correctly, as we do for the memory.

**Comment by mvollmer** [comment]
> Alright, looks like we need to measure the actual load in some other way and then check whether the plot reflects it correctly, as we do for the memory.

Done.

**Comment by stefwalter** [comment]
# Measured CPU 4.60821762914 Wrote TestMetrics-testInternal-FAIL.png Journal database copied to TestMetrics-testInternal-10.111.113.34-FAIL.journal Journal extracted to TestMetrics-testInternal-10.111.113.34-FAIL.log not ok 19 testInternal (check_metrics.TestMetrics) Traceback (most recent call last): File "./verify/check-metrics", line 155, in testInternal self.check_host_metrics() File "./verify/check-metrics", line 109, in check_host_metrics cpu_percentage*0.95, cpu_percentage*1.05, 15, "cpu")); AssertionError: False is not true

**Comment by mvollmer** [comment]
> Measured CPU 4.60821762914 

Hmm, did the time get adjusted by NTP?

**Comment by stefwalter** [comment]
```
CalledProcessError: Command 'pgrep cockpit-pcp' returned non-zero exit status 1 
```

**Comment by mvollmer** [comment]
> CalledProcessError: Command 'pgrep cockpit-pcp' returned non-zero exit status 1 

I think I can explain this.  During calls to `wait`, the browser is stopped and if cockpit-pcp isn't yet running, it will not be started.

Let's just check whether or not cockpit-pcp is running at the end.

**Pull request closed without merging**